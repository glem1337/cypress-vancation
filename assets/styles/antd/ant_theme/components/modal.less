// Modal
@modal-body-padding: 0;
@modal-header-bg: @component-background;
@modal-header-border-color-split: @border-color-split;
@modal-content-bg: @component-background;
@modal-heading-color: @heading-color;
@modal-footer-bg: transparent;
@modal-footer-border-color-split: @border-color-split;
@modal-footer-padding-vertical: 24px;
@modal-footer-padding-horizontal: 24px;
@modal-mask-bg: fade(black, 50%);
