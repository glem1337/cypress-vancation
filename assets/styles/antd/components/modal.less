.ant-modal-wrap {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.ant-modal-content {
  @media (max-width: @screen-md - 1px) {
    border-bottom-right-radius: unset;
    border-bottom-left-radius: unset;
  }
}

.ant-modal-close {
  top: 2px;
}

.ant-modal-close-x {
  color: @gray500;
  transition: color 0.3s;

  &:hover {
    color: @black;
  }

  svg {
    display: none;
  }
}
