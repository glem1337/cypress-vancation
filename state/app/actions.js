import {
  SET_PAGES,
  SET_CURRENT_LOCATION,
  SET_CURRENT_COORDINATES,
  SET_OPEN_GRAPH_DATA,
} from './type';

export const setPages = ({ currentPage, prevPage, isServer }) => ({
  type: SET_PAGES,
  isServer: isServer || false,
  currentPage,
  prevPage,
});

export const setCurrentCoordinates = ({ longitude, latitude }) => ({
  type: SET_CURRENT_COORDINATES,
  longitude,
  latitude,
});

export const setCurrentLocation = (location) => ({
  type: SET_CURRENT_LOCATION,
  location,
});

export const setOpenGraphData = (data) => ({
  type: SET_OPEN_GRAPH_DATA,
  data,
});
