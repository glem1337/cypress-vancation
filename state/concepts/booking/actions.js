import makeFormSubmitAction from 'utils/form/makeFormSubmitAction';

import {
  FETCH_CAMPER_BOOKING_INQUIRY,
  CREATE_CAMPER_BOOKING_INQUIRY,
  SET_CHAT_SECTION_VISIBILITY,
  SET_DETAILS_SECTION_VISIBILITY,
  FETCH_CAMPER_INQUIRIES,
  SET_CAMPER_INQUIRIES_IDS,
  SET_CAMPER_INQUIRIES_PAGE,
  SET_CAMPER_INQUIRIES_TOTAL,
  SET_FILTERS_VISIBILITY,
  SET_ACTIVE_BOOKING_ID,
  FETCH_BOOKING_CHAT_MESSAGES,
  SET_BOOKING_CHAT_MESSAGES_IDS,
  SET_BOOKING_CHAT_MESSAGES_PAGE,
  SET_BOOKING_CHAT_MESSAGES_TOTAL,
} from './types';

export const createCamperBookingInquiry = makeFormSubmitAction(
  CREATE_CAMPER_BOOKING_INQUIRY,
);

export const fetchCamperBookingInquiry = (id) => ({
  type: FETCH_CAMPER_BOOKING_INQUIRY,
  id,
});

export const setChatSectionVisibility = (isVisible) => ({
  type: SET_CHAT_SECTION_VISIBILITY,
  isVisible,
});

export const setDetailsSectionVisibility = (isVisible) => ({
  type: SET_DETAILS_SECTION_VISIBILITY,
  isVisible,
});

export const fetchCamperInquiries = ({
  page = 1,
  perPage = 20,
  userType,
} = {}) => ({
  type: FETCH_CAMPER_INQUIRIES,
  page,
  perPage,
  userType,
});

export const setCamperInquiriesIds = (ids) => ({
  type: SET_CAMPER_INQUIRIES_IDS,
  ids,
});

export const setCamperInquiriesPage = (page) => ({
  type: SET_CAMPER_INQUIRIES_PAGE,
  page,
});

export const setCamperInquiriesTotal = (total) => ({
  type: SET_CAMPER_INQUIRIES_TOTAL,
  total,
});

export const setFiltersVisibility = (isVisible) => ({
  type: SET_FILTERS_VISIBILITY,
  isVisible,
});

export const setActiveBookingId = (id) => ({
  type: SET_ACTIVE_BOOKING_ID,
  id,
});

export const fetchBookingChatMessages = ({
  id = null,
  page = 1,
  perPage = 20,
} = {}) => ({
  type: FETCH_BOOKING_CHAT_MESSAGES,
  id,
  page,
  perPage,
});

export const setBookingChatMessagesIds = (ids) => ({
  type: SET_BOOKING_CHAT_MESSAGES_IDS,
  ids,
});

export const setBookingChatMessagesPage = (page) => ({
  type: SET_BOOKING_CHAT_MESSAGES_PAGE,
  page,
});

export const setBookingChatMessagesTotal = (total) => ({
  type: SET_BOOKING_CHAT_MESSAGES_TOTAL,
  total,
});
