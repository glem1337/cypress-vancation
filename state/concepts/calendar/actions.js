import {
  SET_CURRENT_DATE,
  SET_SETTINGS_VISIBILITY,
  SET_SELECTED_SLOTS,
  CLEAR_SELECTED_SLOTS,
  SET_AVAILABILITY_VISIBILITY,
  SET_AVAILABILITY_MODE,
  CLOSE_RIGHT_SIDEBAR,
  TOGGLE_CALENDAR_FOOTER,
  FIND_EXTERNAL_CALENDAR_IDS,
  SET_EXTERNAL_CALENDAR_IDS,
  REMOVE_EXTERNAL_CALENDAR,
  SET_CUSTOM_DISCOUNT_PERIODS_IDS,
  SET_CUSTOM_NIGHT_RATES_IDS,
  SET_BLOCKED_PERIODS_IDS,
  SET_CUSTOM_MIN_NIGHT_STAY_IDS,
  SET_EXTERNAL_EVENT_IDS,
} from './types';

export const setCurrentDate = (date) => ({
  type: SET_CURRENT_DATE,
  date,
});

export const setCalendarSettingsVisibility = (isVisible) => ({
  type: SET_SETTINGS_VISIBILITY,
  isVisible,
});

export const setAvailabilityVisibility = (isVisible) => ({
  type: SET_AVAILABILITY_VISIBILITY,
  isVisible,
});

export const setAvailabilityMode = (mode) => ({
  type: SET_AVAILABILITY_MODE,
  mode,
});

export const closeRightSidebar = () => ({
  type: CLOSE_RIGHT_SIDEBAR,
});

export const setSelectedSlots = ({ start, end, slots }) => ({
  type: SET_SELECTED_SLOTS,
  start,
  end,
  slots,
});

export const clearSelectedSlots = () => ({
  type: CLEAR_SELECTED_SLOTS,
});

export const toggleCalendarFooter = () => ({
  type: TOGGLE_CALENDAR_FOOTER,
});

export const findExternalCalendarIds = (response) => ({
  type: FIND_EXTERNAL_CALENDAR_IDS,
  response,
});

export const setExternalCalendarIds = (ids) => ({
  type: SET_EXTERNAL_CALENDAR_IDS,
  ids,
});

export const removeExternalCalendar = (id) => ({
  type: REMOVE_EXTERNAL_CALENDAR,
  id,
});

export const setCustomDiscountPeriodIds = (ids) => ({
  type: SET_CUSTOM_DISCOUNT_PERIODS_IDS,
  ids,
});

export const setCustomNightRatesIds = (ids) => ({
  type: SET_CUSTOM_NIGHT_RATES_IDS,
  ids,
});

export const setCustomMinNightStayIds = (ids) => ({
  type: SET_CUSTOM_MIN_NIGHT_STAY_IDS,
  ids,
});

export const setBlockedPeriodIds = (ids) => ({
  type: SET_BLOCKED_PERIODS_IDS,
  ids,
});

export const setExternalEventsIds = (ids) => ({
  type: SET_EXTERNAL_EVENT_IDS,
  ids,
});
