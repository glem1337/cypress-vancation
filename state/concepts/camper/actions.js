import makeFormSubmitAction from 'utils/form/makeFormSubmitAction';

import {
  CREATE_NAME_AND_DESCRIPTION,
  FETCH_CAMPER,
  CREATE_BATCH_CAMPER_PHOTOS,
  CREATE_CAMPER_PHOTO,
  DELETE_CAMPER_PHOTO,
  CREATE_CAMPER,
  FETCH_SPECIFICATIONS,
  CREATE_CAMPER_POLICIES,
  UPDATE_CAMPER_POLICIES,
  CREATE_TRIP_FEES,
  CREATE_CAMPER_PRICING,
  DELETE_CUSTOM_TRIP_FEE,
  CREATE_DELIVERY,
  UPDATE_DELIVERY,
  FETCH_CAMPER_AMENITIES,
  CHECK_PREVIOUS_STEP_COMPLETENESS,
  SET_ESTIMATE_EARNING_DATA,
  CLEAR_ESTIMATE_EARNING_DATA,
  SET_ESTIMATE_EARNING_STATE,
  CREATE_CAMPER_CALENDAR_EXPORT_LINK,
  CREATE_CAMPER_CALENDAR_IMPORT,
  SET_LEAVE_PAGE_METHOD,
  CREATE_CAMPER_AMENITIES,
  CREATE_CUSTOM_DISCOUNT_PERIOD,
  CREATE_CUSTOM_MIN_NIGHT_STAY_PERIOD,
  CREATE_CUSTOM_NIGHT_RATE_PERIOD,
  CREATE_BLOCKED_PERIODS,
  DELETE_BLOCKED_PERIODS,
  DELETE_CAMPER_EXTERNAL_CALENDAR,
  UPDATE_CAMPER_CALENDAR_IMPORT,
  FETCH_OWNER_CAMPERS,
  SET_OWNER_CAMPER_IDS,
  SET_OWNER_CAMPERS_PAGE,
  SET_OWNER_TOTAL,
  SET_OWNER_CAMPERS_STATUS_FILTER,
  SET_FIRST_PORTION_CAMPER_IDS,
  UPDATE_DEFAULT_NIGHT_RATE,
  UPDATE_BASE_DISCOUNTS,
  UPDATE_MIN_NIGHT_STAY,
  UPDATE_CAMPER_SPECIFICATIONS,
  UPDATE_CALENDAR_AVAILABILITY,
  UPDATE_PREPARATION_TIME,
  UPDATE_PICKUP_DROP_OFF_TIME,
  UPDATE_NAME_AND_DESCRIPTION,
  CREATE_INSURANCE_INFO,
  FETCH_CAMPER_CALENDAR,
  CREATE_CAMPER_AMENITY_HEALTH_SAFETIES,
  UPDATE_INSURANCE_INFO,
  UPDATE_TRIP_FEES,
  UPDATE_CAMPER_STATUS,
  DELETE_OWNER_CAMPER_ID,
  DELETE_FIRST_PORTION_CAMPER_ID,
  SET_OWNER_CAMPERS_SEARCH,
  CREATE_CAMPER_RESTRICTIONS,
  DELETE_CAMPER_RESTRICTIONS,
  CREATE_CAMPER_DOCUMENTS,
  DELETE_CAMPER_DOCUMENT,
  DELETE_CAMPER_QUESTION,
  FETCH_VEHICLE_TYPES,
  CREATE_CAMPER_TRAVEL_ACCESSORIES,
  DELETE_CAMPER_CUSTOM_TRAVEL_ACCESSORIES,
  FETCH_CAMPER_TRAVEL_ASSERTIONS,
  FETCH_CAMPER_POLICIES,
  FETCH_CAMPER_FACILITIES,
  FETCH_CAMPER_TRAVEL_EXTENTION,
  FETCH_CAMPER_OWNER,
  FETCH_CAMPER_PRICING_AND_FEES,
  SET_CAMPER_PRICING_AND_FEES_ID,
} from './types';

export const createNameAndDescription = makeFormSubmitAction(CREATE_NAME_AND_DESCRIPTION);

export const updateNameAndDescription = makeFormSubmitAction(UPDATE_NAME_AND_DESCRIPTION);

export const fetchCamper = (id, inclusions) => ({
  type: FETCH_CAMPER,
  id,
  inclusions,
});

export const createBatchCamperPhotos = ({
  camperId, photos, mainPhoto, order, redirectRoute,
}) => ({
  type: CREATE_BATCH_CAMPER_PHOTOS,
  photos,
  camperId,
  mainPhoto,
  order,
  redirectRoute,
});

export const createCamperPhoto = ({ camperId, photo }) => ({
  type: CREATE_CAMPER_PHOTO,
  photo,
  camperId,
});

export const deleteCamperPhoto = (photoId) => ({
  type: DELETE_CAMPER_PHOTO,
  photoId,
});

export const createCamperAction = makeFormSubmitAction(CREATE_CAMPER);

export const fetchSpecificationsAction = () => ({
  type: FETCH_SPECIFICATIONS,
});

export const createCamperPolicies = makeFormSubmitAction(
  CREATE_CAMPER_POLICIES,
);

export const updateCamperPolicies = makeFormSubmitAction(
  UPDATE_CAMPER_POLICIES,
);

export const createTripFees = makeFormSubmitAction(CREATE_TRIP_FEES);

export const updateTripFees = makeFormSubmitAction(UPDATE_TRIP_FEES);

export const createCamperDeliveryAction = makeFormSubmitAction(CREATE_DELIVERY);

export const updateCamperDeliveryAction = makeFormSubmitAction(UPDATE_DELIVERY);

export const createCamperPricing = makeFormSubmitAction(
  CREATE_CAMPER_PRICING,
);

export const deleteCustomTripFee = ({ feeId, camperId }) => ({
  type: DELETE_CUSTOM_TRIP_FEE,
  feeId,
  camperId,
});

export const checkPreviousStepCompleteness = ({ camperId, key, ctx }) => ({
  type: CHECK_PREVIOUS_STEP_COMPLETENESS,
  key,
  camperId,
  ctx,
});

export const setEstimateEarningData = ({ estimatedEarning, iconUrl, name }) => ({
  type: SET_ESTIMATE_EARNING_DATA,
  estimatedEarning,
  iconUrl,
  name,
});

export const clearEstimateEarningData = () => ({
  type: CLEAR_ESTIMATE_EARNING_DATA,
});

export const setEstimateEarningState = (state) => ({
  type: SET_ESTIMATE_EARNING_STATE,
  state,
});

export const createCamperCalendarExportLink = (camperId) => ({
  type: CREATE_CAMPER_CALENDAR_EXPORT_LINK,
  camperId,
});

export const createCamperCalendarImport = makeFormSubmitAction(
  CREATE_CAMPER_CALENDAR_IMPORT,
);

export const fetchCamperAmenities = (inclusions) => ({
  type: FETCH_CAMPER_AMENITIES,
  inclusions,
});

export const setLeavePageMethod = (leavePageMethod) => ({
  type: SET_LEAVE_PAGE_METHOD,
  leavePageMethod,
});

export const createCamperAmenities = makeFormSubmitAction(
  CREATE_CAMPER_AMENITIES,
);

export const createCustomDiscountPeriod = makeFormSubmitAction(CREATE_CUSTOM_DISCOUNT_PERIOD);

export const createCustomMinNightStayPeriod = makeFormSubmitAction(
  CREATE_CUSTOM_MIN_NIGHT_STAY_PERIOD,
);

export const createCustomNightRatePeriod = makeFormSubmitAction(
  CREATE_CUSTOM_NIGHT_RATE_PERIOD,
);

export const createBlockedPeriods = ({ camperId, startDate, endDate }) => ({
  type: CREATE_BLOCKED_PERIODS,
  camperId,
  startDate,
  endDate,
});

export const deleteBlockedPeriods = ({ camperId, startDate, endDate }) => ({
  type: DELETE_BLOCKED_PERIODS,
  camperId,
  startDate,
  endDate,
});

export const deleteCamperExternalCalendar = ({ calendarId, camperId }) => ({
  type: DELETE_CAMPER_EXTERNAL_CALENDAR,
  calendarId,
  camperId,
});

export const createInsuranceInfo = (camperId) => ({
  type: CREATE_INSURANCE_INFO,
  camperId,
});

export const updateInsuranceInfo = makeFormSubmitAction(UPDATE_INSURANCE_INFO);

export const fetchCamperCalendar = ({ camperId, startDate, endDate }) => ({
  type: FETCH_CAMPER_CALENDAR,
  camperId,
  startDate,
  endDate,
});

export const updateCamperCalendarImport = makeFormSubmitAction(
  UPDATE_CAMPER_CALENDAR_IMPORT,
);

export const fetchOwnerCampers = () => ({
  type: FETCH_OWNER_CAMPERS,
});

export const setOwnerCamperIds = camperIds => ({
  type: SET_OWNER_CAMPER_IDS,
  camperIds,
});

export const deleteOwnerCamperId = (camperId) => ({
  type: DELETE_OWNER_CAMPER_ID,
  camperId,
});

export const setFirstPortionCamperIds = camperIds => ({
  type: SET_FIRST_PORTION_CAMPER_IDS,
  camperIds,
});

export const deleteFirstPortionCamperId = camperId => ({
  type: DELETE_FIRST_PORTION_CAMPER_ID,
  camperId,
});

export const setOwnerCampersPage = page => ({
  type: SET_OWNER_CAMPERS_PAGE,
  page,
});

export const setOwnerTotal = total => ({
  type: SET_OWNER_TOTAL,
  total,
});

export const setOwnerCampersStatusFilter = status => ({
  type: SET_OWNER_CAMPERS_STATUS_FILTER,
  status,
});

export const setOwnerCampersSearch = search => ({
  type: SET_OWNER_CAMPERS_SEARCH,
  search,
});

export const updateDefaultNightRate = makeFormSubmitAction(UPDATE_DEFAULT_NIGHT_RATE);

export const updateBaseDiscounts = makeFormSubmitAction(UPDATE_BASE_DISCOUNTS);

export const updateMinNightStay = makeFormSubmitAction(UPDATE_MIN_NIGHT_STAY);

export const updateCamperSpecification = makeFormSubmitAction(UPDATE_CAMPER_SPECIFICATIONS);

export const updateCalendarAvailability = makeFormSubmitAction(UPDATE_CALENDAR_AVAILABILITY);

export const updateCalendarPreparationTime = makeFormSubmitAction(UPDATE_PREPARATION_TIME);

export const updatePickupDropOffTime = makeFormSubmitAction(UPDATE_PICKUP_DROP_OFF_TIME);

export const createCamperAmenityHealthSafeties = makeFormSubmitAction(
  CREATE_CAMPER_AMENITY_HEALTH_SAFETIES,
);

export const updateCamperStatus = ({ status, camperId }) => ({
  type: UPDATE_CAMPER_STATUS,
  status,
  camperId,
});

export const createCamperRestrictions = makeFormSubmitAction(CREATE_CAMPER_RESTRICTIONS);

export const deleteCamperCustomRestrictions = ({ id, camperId, customRestrictionType }) => ({
  type: DELETE_CAMPER_RESTRICTIONS,
  id,
  camperId,
  customRestrictionType,
});

export const createCamperDocuments = makeFormSubmitAction(
  CREATE_CAMPER_DOCUMENTS,
);

export const deleteCamperDocument = ({
  documentId,
  camperId,
}) => ({
  type: DELETE_CAMPER_DOCUMENT,
  documentId,
  camperId,
});

export const deleteCamperQuestion = ({
  questionId,
  camperId,
}) => ({
  type: DELETE_CAMPER_QUESTION,
  questionId,
  camperId,
});

export const fetchVehicleTypes = () => ({
  type: FETCH_VEHICLE_TYPES,
});

export const createCamperTravelAccessories = makeFormSubmitAction(
  CREATE_CAMPER_TRAVEL_ACCESSORIES,
);

export const deleteCamperCustomTravelAccessory = ({ addonId, camperId }) => ({
  type: DELETE_CAMPER_CUSTOM_TRAVEL_ACCESSORIES,
  addonId,
  camperId,
});

export const fetchCamperTravelAssertions = (id) => ({
  type: FETCH_CAMPER_TRAVEL_ASSERTIONS,
  id,
});

export const fetchCamperPolicies = (id) => ({
  type: FETCH_CAMPER_POLICIES,
  id,
});

export const fetchCamperFacilities = (id) => ({
  type: FETCH_CAMPER_FACILITIES,
  id,
});

export const fetchCamperTravelExtention = (id) => ({
  type: FETCH_CAMPER_TRAVEL_EXTENTION,
  id,
});

export const fetchCamperOwner = (id) => ({
  type: FETCH_CAMPER_OWNER,
  id,
});

export const fetchCamperPricingAndFees = ({
  camperId,
  startDate,
  endDate,
}) => ({
  type: FETCH_CAMPER_PRICING_AND_FEES,
  camperId,
  startDate,
  endDate,
});

export const setCamperPricingAndFeesId = (id) => ({
  type: SET_CAMPER_PRICING_AND_FEES_ID,
  id,
});
