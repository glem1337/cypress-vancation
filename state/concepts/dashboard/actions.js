import { SET_MOBILE_MENU_VISIBILITY } from './types';

// eslint-disable-next-line import/prefer-default-export
export const setMobileMenuVisibility = isVisible => ({
  type: SET_MOBILE_MENU_VISIBILITY,
  isVisible,
});
