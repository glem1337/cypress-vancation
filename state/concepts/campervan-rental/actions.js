import { CAMPER_INCLUSION } from 'constants/camper';
import { LANDING_TYPE } from 'constants/campervanRentals';
import { SEARCH_RESULTS_RADIUS } from 'constants/searchDestinations';

import {
  FETCH_STATE,
  FETCH_STATE_LOCATION,
  MAP_LOCATION_SLUG_TO_ID,
  MAP_STATE_SLUG_TO_ID,
  FETCH_NEARBY_DESTINATIONS,
  SET_NEARBY_DESTINATIONS_IDS,
  FETCH_EPICENTER_LOCATIONS,
  FETCH_HOME_STATES,
  SET_SHOW_ALL_STATES,
  SET_SLIDE_FAVORITE_DESTINATION,
  SET_FAVORITE_TOTAL,
  FETCH_FAVORITE_DESTINATION,
  FETCH_CAMPERS,
  SET_CAMPER_IDS,
  SET_CAMPERS_PAGE,
  SET_CAMPERS_TOTAL,
  RESET_CAMPERS_DATA,
} from './types';

export const fetchState = ({
  state,
  inclusions = 'seo_info,top_location_landings,fun_facts',
}) => ({
  type: FETCH_STATE,
  state,
  inclusions,
});

export const fetchStateLocation = ({
  state,
  location,
  inclusions = 'state_landing,seo_info,fun_facts',
}) => ({
  type: FETCH_STATE_LOCATION,
  state,
  location,
  inclusions,
});

export const mapStateSlugToId = ({ id, slug }) => ({
  type: MAP_STATE_SLUG_TO_ID,
  id,
  slug,
});

export const mapLocationSlugToId = ({ id, slug }) => ({
  type: MAP_LOCATION_SLUG_TO_ID,
  id,
  slug,
});

export const fetchNearbyDestinations = ({
  latitude,
  longitude,
  count = 12,
  landingType = LANDING_TYPE.LOCATION_LANDING,
  excludedId,
}) => ({
  type: FETCH_NEARBY_DESTINATIONS,
  latitude,
  longitude,
  count,
  landingType,
  excludedId,
});

export const setNearbyDestinationsIds = (ids) => ({
  type: SET_NEARBY_DESTINATIONS_IDS,
  ids,
});

export const fetchEpicenterLocationsAction = () => ({
  type: FETCH_EPICENTER_LOCATIONS,
});

export const fetchHomeStatesAction = () => ({
  type: FETCH_HOME_STATES,
});

export const setShowAllStatesAction = (state) => ({
  type: SET_SHOW_ALL_STATES,
  state,
});

export const setSlideFavoriteDestinationsAction = (slide) => ({
  type: SET_SLIDE_FAVORITE_DESTINATION,
  slide,
});

export const setFavoriteTotalAction = (total) => ({
  type: SET_FAVORITE_TOTAL,
  total,
});

export const fetchFavoriteDestinationsAction = ({ pageNumber, pageSize }) => ({
  type: FETCH_FAVORITE_DESTINATION,
  pageNumber,
  pageSize,
});

export const fetchCampers = ({
  latitude,
  longitude,
  startDate,
  endDate,
  page = 1,
  perPage = 20,
  radius = SEARCH_RESULTS_RADIUS,
  inclusions = CAMPER_INCLUSION.SPECIFICATIONS_DETAILS,
} = {}) => ({
  type: FETCH_CAMPERS,
  latitude,
  longitude,
  startDate,
  endDate,
  page,
  perPage,
  radius,
  inclusions,
});

export const setCamperIds = (ids) => ({
  type: SET_CAMPER_IDS,
  ids,
});

export const setCampersTotal = (total) => ({
  type: SET_CAMPERS_TOTAL,
  total,
});

export const setCampersPage = (page) => ({
  type: SET_CAMPERS_PAGE,
  page,
});

export const resetCampersData = () => ({
  type: RESET_CAMPERS_DATA,
});
