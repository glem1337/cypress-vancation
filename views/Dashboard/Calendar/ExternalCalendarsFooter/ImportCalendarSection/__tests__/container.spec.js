import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import diveTo from 'utils/testHelpers/diveToEnzyme';
import { showModal } from 'state/modal/actions';

import ImportCalendarSectionWrapper, {
  ImportCalendarSectionContainer,
} from '../container';

jest.mock('state/concepts/calendar/selectors', () => ({
  externalCalendarIdsSelector: jest.fn(() => (['id'])),
}));

describe('ImportCalendarSection container tests', () => {
  const store = configureStore()({});
  store.dispatch = jest.fn();

  const props = {
    store,
    showModal: jest.fn(),
    camperId: 'camperId',
  };

  const wrapper = shallow(<ImportCalendarSectionWrapper {...props} />);
  const container = diveTo(wrapper, ImportCalendarSectionContainer);
  const instance = container.instance();

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('test `showImportModal` instance method', () => {
    instance.showImportModal();

    expect(store.dispatch).toHaveBeenCalledWith(
      showModal({
        modalType: 'CALENDAR_IMPORT_MODAL',
        modalProps: {
          camperId: 'camperId',
        },
      }),
    );
  });
});
