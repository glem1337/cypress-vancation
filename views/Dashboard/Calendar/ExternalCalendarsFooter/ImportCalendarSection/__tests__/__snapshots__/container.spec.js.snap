// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportCalendarSection container tests matches snapshot 1`] = `
<ImportCalendarSection
  camperId="camperId"
  onShowImportModal={[Function]}
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
